wallet/src/main/scala/org/bitcoins/wallet/internal/FundTransactionHandling.scala

```diff
@@ -0,0 +1,127 @@
+package org.bitcoins.wallet.internal
+
+import org.bitcoins.core.config.BitcoinNetwork
+import org.bitcoins.core.crypto.Sign
+import org.bitcoins.core.protocol.transaction.{Transaction, TransactionOutput}
+import org.bitcoins.core.wallet.builder.BitcoinTxBuilder
+import org.bitcoins.core.wallet.fee.FeeUnit
+import org.bitcoins.keymanager.bip39.BIP39KeyManager
+import org.bitcoins.wallet.WalletLogger
+import org.bitcoins.wallet.api.{AddressInfo, CoinSelector, LockedWalletApi}
+import org.bitcoins.wallet.models.{AccountDb, SpendingInfoDb}
+
+import scala.concurrent.Future
+
+trait FundTransactionHandling extends WalletLogger { self: LockedWalletApi =>
+
+  def fundRawTransaction(
+      destinations: Vector[TransactionOutput],
+      feeRate: FeeUnit): Future[Transaction] = {
+    for {
+      account <- getDefaultAccount()
+      funded <- fundRawTransaction(destinations = destinations,
+                                   feeRate = feeRate,
+                                   fromAccount = account)
+    } yield funded
+  }
+
+  def fundRawTransaction(
+      destinations: Vector[TransactionOutput],
+      feeRate: FeeUnit,
+      fromAccount: AccountDb): Future[Transaction] = {
+    val txBuilderF =
+      fundRawTransactionInternal(destinations = destinations,
+                                 feeRate = feeRate,
+                                 fromAccount = fromAccount,
+                                 keyManagerOpt = None)
+    txBuilderF.flatMap(_.unsignedTx)
+  }
+
+  /** This returns a [[BitcoinTxBuilder]] that can be used to generate a unsigned transaction with [[BitcoinTxBuilder.unsignedTx unsignedTx]]
+    * which can be used with signing, or you can just directly call [[BitcoinTxBuilder.sign sign]] to sign the transaction with this instance
+    * of [[BitcoinTxBuilder]]
+    *
+    * If you pass in a [[org.bitcoins.keymanager.KeyManager]], the [[org.bitcoins.core.wallet.utxo.UTXOSpendingInfo.signers signers]]
+    * will be populated with really signers that can be used to produce valid [[org.bitcoins.core.crypto.ECDigitalSignature signatures]]
+    *
+    * If you do not pass in a key manager, the transaction will contain [[org.bitcoins.core.protocol.script.EmptyScriptSignature EmptyScriptSignature]]
+    * */
+  private[wallet] def fundRawTransactionInternal(
+      destinations: Vector[TransactionOutput],
+      feeRate: FeeUnit,
+      fromAccount: AccountDb,
+      keyManagerOpt: Option[BIP39KeyManager]): Future[BitcoinTxBuilder] = {
+    val utxosF = listUtxos()
+    val changeAddrF = getNewChangeAddress(fromAccount)
+    val selectedUtxosF = for {
+      walletUtxos <- utxosF
+      change <- changeAddrF
```

Comment:
This is never used, should be fine to remove, the change address shouldn't matter when we are selecting the funding utxos